### PR TITLE
[EVM-Equivalent-Yul] Add mload, mstore and mstore8

### PR DESCRIPTION
**Pull request by ManuelBilbao**
## What ❔

This PR adds `mload`, `mstore` and `mstore8` tests.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

File: core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs [hunk 1]
```diff
@@ -627,6 +627,237 @@ fn test_basic_signextend_vectors() {
...
+                hex::decode("51").unwrap(),
+                // push0
+                hex::decode("5F").unwrap(),
+                // sstore
+                hex::decode("55").unwrap(),
+            ]
+            .into_iter()
+            .concat()
+        ),
+        16_711_680.into()
```
Comment (gianbelinche) [inline core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs:706]: Why this number?

File: contracts
Comment (gianbelinche) [inline contracts:1]: I think we should push this, its the git submodule

**Review by gianbelinche — CHANGES_REQUESTED** [review]
Everything else is ok!

File: core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs (hunk 1 above)
Comment (fborello-lambda) [reply core/lib/multivm/src/versions/vm_latest/tests/evm_simulator.rs:706]: This [playground](https://www.evm.codes/playground?unit=Wei&codeType=Mnemonic&code=%27PUSH3%20125985%5CnPOP%27_) shows why 16_711_680 is used. 16711680 == 0xFF0000

**Pull request merged**